dynomite-derive/src/lib.rs

```diff
@@ -228,12 +260,59 @@ fn expand_item(ast: DeriveInput) -> syn::Result<impl ToTokens> {
     }
 }
 
+fn make_dynomite_attributes(
+    name: &Ident,
+    fields: &[Field],
+) -> syn::Result<impl ToTokens> {
+    let item_fields = fields.iter().map(ItemField::new).collect::<Vec<_>>();
+    // impl ::dynomite::FromAttributes for Name
+    let from_attribute_map = get_from_attributes_trait(name, &item_fields)?;
+    // impl From<Name> for ::dynomite::Attributes
+    let to_attribute_map = get_to_attribute_map_trait(name, &item_fields)?;
+    // impl Attribute for Name (these are essentially just a map)
+    let attribute = quote!(::dynomite::Attribute);
+    let impl_attribute = quote! {
+        impl #attribute for #name {
```

Comment:
we need to derive Attribute since embedded structs within Items need to be viewed as an Attribute. It turns out that makes this impl quite simple since its basically just a map of attributes anyway!